subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskIntegrationTest.groovy

```diff
@@ -92,19 +92,49 @@ public class TestTaskIntegrationTest extends AbstractIntegrationSpec {
 
     }
 
+    def "test task do not hang if maxParallelForks is greater than max-workers"() {
+        given:
+        def maxWorkers = Runtime.runtime.availableProcessors()
```

Comment:
What happened if you manually set max workers higher than the available processors (i.e. `StartParameter.getMaxWorkerCount() > Runtime.runtime.availableProcessors()`)

Is that case fine?